src/util.rs

```diff
@@ -58,13 +52,28 @@ pub fn find_file(file: &str, subdir: Option<&str>, extension: Option<&str>) -> O
         usr_share_path.push(subdir);
     }
     usr_share_path.push(&file);
-    if usr_share_path.exists() {
-        return Some(usr_share_path);
+    if let Some(file) = exists_with_extension(&usr_share_path, extension) {
+        return Some(file);
     }
 
     None
 }
 
+fn exists_with_extension<'a>(file: &'a Path, extension: Option<&str>) -> Option<PathBuf> {
```

Comment:
Not sure about the name, almost wants to be something like `exists_with_or_without_extension` but less of a mouthful

Reply:
Perhaps `exists_with_opt_extension`?

Reply:
Thanks. Feels a bit like it's duplicating the signature but it's better than anything I can come up with.